### DISCUSSION TITLE
Question about GFM units

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#436] [Q&A] [answered]
Dear Dr.Cui and Jinning,
I have two questions as following:
1. When we modify three traditional generators into GFM units, in one case, the power rotor angle of some buses is lower than -2000 degrees, and the voltage drops to a negative value. Is this normal?
2. In order to learn the influences of key parameters, such as moment of inertia, damping coefficient, droop coefficient stability of control systems, we want to modify some parameters of REGF1 and REGF2. However, we have not found these in the parameters list of these two models.
Best wishes! Looking forward to your answer!

**Comment by jinningwang** [comment, marked as answer]
Hello @xiaoheimai95

1. I don't think this is reasonable. You can try to modify the initial value of voltage and angle to see if there will be any difference.
2. As you said, there does not exist parameters `M`, `D`, `R` in the model `REGF1` and `REGF2`. I failed to find any online materials for these two models from my end. For now, you can try to read the source code of these models to understand how to control these models. You can try to plot the control diagram according to the source code.

Hantao @cuihantao 
Are there any public materials for `REGF1` and `REGF2`?

Regards,
Jinning

**Reply by xiaoheimai95** [reply]
Thanks！

**Reply by jinningwang** [reply]
@xiaoheimai95

Would you mind sharing the control diagram if you finally get some?

Regards,
Jinning

**Reply by cuihantao** [reply]
[Ramasubramanian et al. - 2021 - SUBJECT PROPOSAL FOR SUITE OF GENERIC GRID FORMIN.pdf](https://github.com/CURENT/andes/files/11149832/Ramasubramanian.et.al.-.2021.-.SUBJECT.PROPOSAL.FOR.SUITE.OF.GENERIC.GRID.FORMIN.pdf)
The implemented draft models can be found here.

**Reply by xiaoheimai95** [reply]
Thanks @cuihantao ! And also are there any documents about REGCV1 and REGCV2?

**Reply by jinningwang** [reply]
There is a paper by Dr. Hantao Cui that might have diagrams https://ieeexplore.ieee.org/document/9777891
